app/helpers/questions_helper.rb

```diff
@@ -64,4 +64,13 @@ def remove_questions(questions)
     end
     updated_questions_to_ask
   end
+
+  def add_questions(questions, after_question)
+    updated_questions_to_ask = questions_to_ask
+    questions.each do |question|
+      updated_questions_to_ask.insert(updated_questions_to_ask.index(after_question) + 1, question) unless updated_questions_to_ask.include?(question)
+      after_question = question
+    end
+    updated_questions_to_ask
```

Comment:
Would it be possible to do something like this here?

```ruby
qs = questions_to_ask
index = questions_to_ask.index(after_question)
questions_to_ask.slice(0..index) + questions + questions_to_ask.slice(index..)
```

E.g.

```ruby
alphabet = ('a'..'g').to_a
alphabet.slice(0,3) + %w(x y z) + alphabet.slice(3..)
# => ["a", "b", "c", "x", "y", "z", "d", "e", "f", "g"]
```

Reply:
Having never seen that syntax before, it's quite hard to understand and would need a couple of comments as to what it's doing, I think.

Reply:
Realised I can condense this down into:
```  def add_questions(questions, after_question)
    index = questions_to_ask.index(after_question) + 1
    questions_to_ask
      .dup
      .insert(index, questions)
      .flatten
      .uniq
  end```

The `dup` is required so we don't modify the value of `questions_to_ask` since `insert` would otherwise change it's value.